pkg/model/api/v1/rolebinding.go

```diff
@@ -0,0 +1,227 @@
+		return err
+	}
+	*s = tmp
+	return nil
+}
+
+func (s *Subject) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	var tmp Subject
+	type plain Subject
+	if err := unmarshal((*plain)(&tmp)); err != nil {
+		return err
+	}
+	if err := (&tmp).validate(); err != nil {
+		return err
+	}
+	*s = tmp
+	return nil
+}
+
+func (s *Subject) validate() error {
+	if s.Kind != KindUser {
+		return fmt.Errorf("invalid kind: %q for a Subject kind", s.Kind)
+	}
+	if len(s.Name) == 0 {
+		return fmt.Errorf("subject name cannot be empty")
+	}
+	return nil
+}
+
+type RoleBindingSpec struct {
+	// Name of the Role or GlobalRole concerned by the role binding (metadata.name)
+	Role string `json:"name" yaml:"name"`
+	// Subjects that will inherit permissions from the role
+	Subjects []Subject `json:"subjects" yaml:"subjects"`
+}
+
+func (r *RoleBindingSpec) UnmarshalJSON(data []byte) error {
+	var tmp RoleBindingSpec
+	type plain RoleBindingSpec
+	if err := json.Unmarshal(data, (*plain)(&tmp)); err != nil {
+		return err
+	}
+	if err := (&tmp).validate(); err != nil {
+		return err
+	}
+	*r = tmp
+	return nil
+}
+
+func (r *RoleBindingSpec) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	var tmp RoleBindingSpec
+	type plain RoleBindingSpec
+	if err := unmarshal((*plain)(&tmp)); err != nil {
+		return err
+	}
+	if err := (&tmp).validate(); err != nil {
+		return err
+	}
+	*r = tmp
+	return nil
+}
+
+func (r *RoleBindingSpec) validate() error {
+	if len(r.Role) == 0 {
+		return fmt.Errorf("spec cannot be empty")
+	}
+	return nil
+}
+
+// GlobalRoleBinding is the struct representing the roleBinding shared to everybody.
+type GlobalRoleBinding struct {
+	Kind     Kind            `json:"kind" yaml:"kind"`
+	Metadata Metadata        `json:"metadata" yaml:"metadata"`
+	Spec     RoleBindingSpec `json:"spec" yaml:"spec"`
+}
+
+func (g *GlobalRoleBinding) UnmarshalJSON(data []byte) error {
+	var tmp GlobalRoleBinding
+	type plain GlobalRoleBinding
+	if err := json.Unmarshal(data, (*plain)(&tmp)); err != nil {
+		return err
+	}
+	if err := (&tmp).validate(); err != nil {
+		return err
+	}
+	*g = tmp
+	return nil
+}
+
+func (g *GlobalRoleBinding) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	var tmp GlobalRoleBinding
+	type plain GlobalRoleBinding
+	if err := unmarshal((*plain)(&tmp)); err != nil {
+		return err
+	}
+	if err := (&tmp).validate(); err != nil {
+		return err
+	}
+	*g = tmp
+	return nil
+}
+
+func (g *GlobalRoleBinding) validate() error {
+	if g.Kind != KindGlobalRoleBinding {
+		return fmt.Errorf("invalid kind: %q for a GlobalRoleBinding type", g.Kind)
+	}
+	if reflect.DeepEqual(g.Spec, RoleBindingSpec{}) {
+		return fmt.Errorf("spec cannot be empty")
+	}
+	return nil
+}
+
+func (g *GlobalRoleBinding) GetMetadata() modelAPI.Metadata {
+	return &g.Metadata
+}
+
+func (g *GlobalRoleBinding) GetKind() string {
+	return string(g.Kind)
+}
+
+func (g *GlobalRoleBinding) GetRoleBindingSpec() RoleBindingSpec {
```

Comment:
do you plan to add some generic method ? (I didn't see them for the moment)

Reply:
I don't known, I will remove it for now